server/src/main/java/org/opensearch/rest/pagination/PaginationStrategy.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.pagination;
+
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.metadata.IndexMetadata;
+
+import java.util.Base64;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Interface to be implemented by any strategy getting used for paginating rest responses.
+ *
+ * @opensearch.internal
+ */
+public interface PaginationStrategy<T> {
+
+    String DESCENDING_SORT_PARAM_VALUE = "descending";
+    String INCORRECT_TAINTED_NEXT_TOKEN_ERROR_MESSAGE =
+        "Parameter [next_token] has been tainted and is incorrect. Please provide a valid [next_token].";
+
+    /**
+     *
+     * @return Base64 encoded string, which can be used to fetch next page of response.
+     */
+    PaginatedQueryResponse getPaginatedQueryResponse();
+
+    /**
+     *
+     * @return List of elements fetched corresponding to the store and token received by the strategy.
+     */
+    List<T> getElementsFromRequestedToken();
+
+    /**
+     *
+     * Utility method to get list of indices sorted by their creation time.
+     */
+    static List<IndexMetadata> getListOfIndicesSortedByCreateTime(final ClusterState clusterState, String sortOrder) {
+        return clusterState.metadata().indices().values().stream().sorted((metadata1, metadata2) -> {
```

Comment:
I have defined a new comparator, but still kept only a single method. Since it's just one comparison that changes, thought it's better to avoid code duplication. But still open to defining two separate methods, if you feel that would bring out more clarity